### ISSUE TITLE
CA1062 false positive with explicit cast

### ISSUE THREAD

**Issue by RobSiklos** [dotnet/roslyn-analyzers#2630]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.4-beta1.final

#### Diagnostic ID

[CA1062](https://docs.microsoft.com/en-ca/visualstudio/code-quality/ca1062-validate-arguments-of-public-methods)

#### Repro steps
Analyze the following code:

```csharp
public static class Class1
{
    public static void DoSomething(object input)
    {
        // *** This line still generates a CA1062 error. ***
        Bar(input);

        // Almost the same thing, this time no CA1062 (as expected).
        Bar2(input);
    }

    private static void Bar(object input)
    {
        if (input is Version)
        {
            var version = (Version)input;
            version.ToString();
        }
    }

    private static void Bar2(object input)
    {
        if (input is Version version)
        {
            version.ToString();
        }
    }
}
```

#### Expected behavior
No warnings.

#### Actual behavior
CA1062 when calling `Bar(input);`

**Comment by mavasani** [comment]
Note to self: Predicate analysis needs to handle is pattern expressions in conditionals.

**Comment by RobSiklos** [comment]
Verified fixed in 2.9.4-beta2.19326.1

Thanks!!
